### PR TITLE
Sqrt newton

### PR DESCRIPTION
**Pull request by meszenab**
Newton's method for sqrt

### CODE REVIEW

File: course5/homework2/newton/newton.cc [hunk 1]
```diff
@@ -0,0 +1,59 @@
+#include <iostream>
+#include <string>
+#include <time.h>
+
+// Newton method of sqrt. The stack frame was 48 for Debug mode and 64 for Release mode.
+
+int i = 0;
```
Comment (maciekgajewski) [inline course5/homework2/newton/newton.cc]: Didn't I explicitly asked in the alignements not to use global variables?
Maybe I forgot...
Anyway, try again, without global variables!

File: course5/homework2/newton/newton.cc [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    if (size > 0)
+    {
+        std::cout << "stack grows downwards" << std::endl;
+    }
+    else if (size < 0)
+    {
+        std::cout << "stack grows upwards" << std::endl;
+    }
+
+    std::cout << "calculation time: " << t << std::endl;
```
Comment (maciekgajewski) [inline course5/homework2/newton/newton.cc]: It isn't calculation time; it's "calculation and printing time"

File: course5/homework2/newton/newton.cc [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    const float y = std::stof(argv[1]); // input number
+    x = std::stof(argv[2]); //initial guess
+    const int it = std::stoi(argv[3]); // number of iterations
+
+    clock_t t = clock();
+    std::cout << "sqrt of " << argv[1] << " is " << function(y, it) << std::endl;
+    t = clock() - t;
+
+    std::cout << "depth" << "=" << i + 1 << std::endl;
+    int size = (start - end)/i;
```
Comment (maciekgajewski) [inline course5/homework2/newton/newton.cc]: So, any conclusions? Hoe big is the stack frame? What is the impact of the build type on the stack frame size?

**Review by maciekgajewski — COMMENTED** [review]
Try without using global variables.
Also, I don't think the number of iterations should be part of the interface.

**Review by maciekgajewski — CHANGES_REQUESTED** [review]
Try without using global variables.
Also, I don't think the number of iterations should be part of the interface.

File: course5/homework2/newton/newton.cc (hunk 3 above)
Comment (meszenab) [reply course5/homework2/newton/newton.cc]: The stack frame was 48 for debug mode and bigger (64) in release mode

File: course5/homework2/newton/newton.cc [hunk 4]
```diff
@@ -0,0 +1,58 @@
+#include <iostream>
+#include <string>
+
+// Newton method of sqrt. The stack frame was 48 for Debug mode and 64 for Release mode.
+
+bool* start = nullptr;
```
Comment (maciekgajewski) [inline course5/homework2/newton/newton.cc]: These are still global. And you don't need bool. To store pure, untyped address just use void*

File: course5/homework2/newton/newton.cc [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+bool* end = nullptr;
+
+float function( float y, int it, float& rx, int& ri)
```
Comment (maciekgajewski) [inline course5/homework2/newton/newton.cc]: "function" is not the best name for a function calculating square root. Make your API such so you'd enjoy using it.

File: course5/homework2/newton/newton.cc [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+int main(int argc, char** argv)
+{
+    const float y = std::stof(argv[1]); // input number
+    float x = std::stof(argv[2]); //initial guess
+    const int it = 1000;
+
+    int i = 0;
+
+    int& ri = i;
+    float& rx = x;
```
Comment (maciekgajewski) [inline course5/homework2/newton/newton.cc]: you don't need these

**Review by maciekgajewski — CHANGES_REQUESTED** [review]
You still have global variables in there

**Pull request closed without merging**